lib/validation/validators.js

```diff
@@ -153,18 +153,29 @@ const validatePositiveNumber = (val, name) => {
 	return val;
 };
 
-const validateStVarOrPositiveNumber = (val, name) => {
+const validateNumber = (val, name) => {
+	if (!Number.isFinite(val)) {
+		throwError(name + ' should be number');
+	}
+
+	return val;
+};
+
+const createStVarOrNumberValidator = (onlyPositiveNumber = false) => (val, name) => {
 	if (!['number', 'string'].includes(typeof val)) {
 		throwError(name + ' should be suitest configuration variable or number');
 	} else if (typeof val === 'number') {
-		validatePositiveNumber(val, name);
+		(onlyPositiveNumber ? validatePositiveNumber : validateNumber)(val, name);
 	} else if (typeof val === 'string' && !/(<%.+%>)/.test(val)) {
```

Comment:
btw, regexp will also accept `A<%var%>B`, this is definitely wrong.
There is also `1<%var%>0` which looks very weird. Please pay attention

Reply:
fixed